### PR TITLE
Add line-gradient example to docs

### LINKED ISSUE #6630: add line-gradient example to docs
🌈👩‍🏫

cc @lbud

### PR DESCRIPTION
**Pull request by ryanhamley**
screen shot 2018-05-22 at 3 13 00 pm
## Launch Checklist

 - [x] briefly describe the changes in this PR
    - Adds example for using `line-gradient` paint property
 - [x] manually test the debug page

Closes #6630

### CODE REVIEW

File: docs/pages/example/line-gradient.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        source: 'line',
+        id: 'line',
+        paint: {
+            'line-color': 'red',
+            'line-width': 14,
+            'line-gradient': [
+                'interpolate',
+                ['linear'],
+                ['line-progress'],
+                0, "rgba(0, 0, 255, 0)",
```
Comment (mollymerp) [inline docs/pages/example/line-gradient.html]: any reason why this can't be "blue" ?

File: docs/pages/example/line-gradient.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    zoom: 12,
+});
+
+var geojson = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "properties": {},
+        "geometry": {
+            "coordinates": [[-77.044211, 38.852924 ], [-77.045659, 38.860158 ], [-77.044232, 38.862326 ], [-77.040879, 38.865454 ], [-77.039936, 38.867698 ], [-77.040338, 38.86943 ], [-77.04264, 38.872528 ], [-77.03696, 38.878424 ], [-77.032309, 38.87937 ], [-77.030056, 38.880945 ], [-77.027645, 38.881779 ], [-77.026946, 38.882645 ], [-77.026942, 38.885502 ], [-77.028054, 38.887449 ], [-77.02806, 38.892088 ], [-77.03364, 38.892108 ], [-77.033643, 38.899926 ] ],
```
Comment (mollymerp) [inline docs/pages/example/line-gradient.html]: format so each coordinate pair is on its own line for readability (like https://www.mapbox.com/mapbox-gl-js/example/geojson-line/), and maybe include a comment that explains that line-gradient can only be used with geojson sources.

File: docs/pages/example/line-gradient.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "type": "LineString"
+        }
+    }]
+};
+
+map.on('load', function () {
+    map.addSource('line', {
+        type: 'geojson',
+        lineMetrics: true,
```
Comment (mollymerp) [inline docs/pages/example/line-gradient.html]: comment here indicating that `lineMetrics: true` is necessary to use a source in a line-gradient layer

**Comment by andrewharvey** [comment]
@ryanhamley `line-gradient` was added for 0.45.0[1] which means it can be merged against the `mb-pages` branch so the example becomes live now rather than waiting for the next release [2].

I'm not sure what to do now, but it would be great to get this PR applied to `mb-pages` so it's live.

[1] https://github.com/mapbox/mapbox-gl-js/blob/master/CHANGELOG.md
[2] https://github.com/mapbox/mapbox-gl-js/blob/master/docs/README.md#committing-and-publishing-documentation

**Comment by ryanhamley** [comment]
Ah, thanks for the suggestion @andrewharvey. I can probably cherry-pick this commit to the `mb-pages` branch so it goes live.

**Comment by ryanhamley** [comment]
This example was added to `mb-pages` by #6738.

**Pull request merged**